src/parquet/util/cpu-info.cc

```diff
@@ -148,7 +148,7 @@ void CpuInfo::Init() {
 #endif
 
   if (max_mhz != 0) {
-    cycles_per_ms_ = max_mhz * 1000;
+    cycles_per_ms_ = static_cast<int64_t>(max_mhz) * 1000;
   } else {
     cycles_per_ms_ = 1000000;
```

Comment:
it doesn't, have created [ticket](https://issues.apache.org/jira/browse/PARQUET-1009)